pysilcam/tracking/silcamtracker.py

```diff
@@ -0,0 +1,418 @@
+                                  bad_lighting_limit=None,
+                                  real_time_stats=False)
+
+    def generator_tracker(self, datapath=None):
+        for f in tqdm(self.files):
+
+            img = silcam_load(f)
+
+            if np.ndim(img) == 3:
+                img = np.min(img, axis=2)
+            imx, imy = np.shape(img)
+            imc = np.zeros((imx, imy, 3), dtype=np.uint8())
+            imc[:, :, 0] = img
+            imc[:, :, 1] = img
+            imc[:, :, 2] = img
+            timestamp = pd.to_datetime(os.path.splitext(os.path.split(f)[-1])[0][1:])
+            yield timestamp, imc
+
+    def process(self):
+        PIX_SIZE = self.PIX_SIZE
+        MIN_LENGTH = self.MIN_LENGTH
+        GOOD_FIT = self.cross_correlation_threshold
+        DATAFILE = self.DATAFILE
+        track_length_limit = self.track_length_limit
+
+        if DATAFILE == '':
+            print('DATAFILE not specified')
+            print('---- END ----')
+            return
+
+        # setup HDF5 file and metadata
+        with h5py.File(DATAFILE + '.h5', "a") as HDF5File:
+            meta = HDF5File.require_group('Meta')
+            meta.attrs['Modified'] = str(pd.datetime.now())
+            meta.attrs['DatasetName'] = 'not implemented'
+            HDF5File.create_group("Tracking")
+
+        tracks = pd.DataFrame()
+        tracks.index.name = 'UPID'
+        UPID = -1
+
+        print('* Tracking....')
+
+        img1, t1 = self.load_image()
+
+        i = 0
+        while True:
+
+            try:
+                img2, t2 = self.load_image()
+            except Exception:
+                break
+
+            i += 1
+
+            try:
+                X, Y, ecd, length, width, im_plot = get_vect(img1, img2,
+                                                             PIX_SIZE, MIN_LENGTH, GOOD_FIT,
+                                                             thresh=self.THRESHOLD,
+                                                             ecd_tolerance=self.ecd_tolerance)
+            except ValueError:
+                print('  Error getting vectors')
+                continue
+
+            if len(X[0]) == 0:
+                continue
+
+            for p in range(len(X[0])):
+                UPID += 1
+                tracks.loc[UPID, 'particle'] = int(p)
+                tracks.loc[UPID, 't1'] = t1
+                tracks.loc[UPID, 't2'] = t2
+                tracks.loc[UPID, 'x1'] = X[0][p]
+                tracks.loc[UPID, 'x2'] = X[1][p]
+                tracks.loc[UPID, 'y1'] = Y[0][p]
+                tracks.loc[UPID, 'y2'] = Y[1][p]
+                tracks.loc[UPID, 'ecd'] = ecd[p]
+                tracks.loc[UPID, 'length'] = length[p]
+                tracks.loc[UPID, 'width'] = width[p]
+
+            tracks = match_last_pair(tracks)
+            with pd.HDFStore(DATAFILE + '.h5', "a") as fh:
+                tracks.to_hdf(fh, 'Tracking/data', mode='r+')

```

Comment:
Mode specification in to_hdf superceded by HDFStore (and also different).

Reply:
done

Reply:
done